File: app/src/main/java/com/projects/ebookapp/SplashScreen.java
package com.projects.ebookapp;

import android.content.Intent;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.projects.ebookapp.login.LoginActivity;

public class SplashScreen extends AppCompatActivity {
    private static final int SPLASH_DELAY_TIME = 950;
    private final Handler mHandler = new Handler();
    private final Launcher mLauncher = new Launcher();

    @Override
    protected void onStart() {
        super.onStart();
        mHandler.postDelayed(mLauncher, SPLASH_DELAY_TIME);
    }

    @Override
    protected void onStop() {
        mHandler.removeCallbacks(mLauncher);
        super.onStop();
    }

    private class Launcher implements Runnable {
        @Override
        public void run() {
            launch();
        }

        private void launch() {
            if (!isFinishing()) {
                startActivity(new Intent(SplashScreen.this, LoginActivity.class));
                finish();
            }
        }
    }
}

File: app/src/main/java/com/projects/ebookapp/SplashScreen.kt
package com.projects.ebookapp

import android.content.Intent
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.projects.ebookapp.login.LoginActivity

class SplashScreen : AppCompatActivity() {
    private val mHandler = Handler()
    private val mLauncher = Launcher()
    override fun onStart() {
        super.onStart()
        mHandler.postDelayed(mLauncher, SPLASH_DELAY_TIME.toLong())
    }

    override fun onStop() {
        mHandler.removeCallbacks(mLauncher)
        super.onStop()
    }

    private inner class Launcher : Runnable {
        override fun run() {
            launch()
        }

        private fun launch() {
            if (!isFinishing) {
                startActivity(Intent(this@SplashScreen, LoginActivity::class.java))
                finish()
            }
        }
    }

    companion object {
        private const val SPLASH_DELAY_TIME = 950
    }
}